src_blog/_posts/Grain-Formatter.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Introducing the Grain Formatter
+subtitle: Bringing consistency and pretty code to Grain
+date: 2021-08-31 12:00:00
+author: Marcus Roberts
+cover: /blog/photos/cover/marie-bellando-mitjans-cBChXxQqEKM-unsplash.jpg
+coverAttribution: Marie Bellando-Mitjans / Unsplash
+tags: Tooling, Developer Experience, 
+---
+
+For any modern programming language to succeed it needs to provide developers the support tools they have become used to – code editor extensions, IDE support and in recent years code formatting or prettier tools.
+
+The acceptance of automatic code formatting tools as providing the “one true style” for development in a language seems to have really caught on with the release of Prettier for Javascript.   More than a linter that reports on how well a piece of code adheres to style and best practices, a code formatter rewrites a piece of code in the approved style.
+
+This really helps codebases and teams in several ways.   It reduces arguments at code review time as the correct style is guaranteed.  It helps new members of the team learn the coding style, it keeps a very consistent code base that’s easy to read by everyone.   My personal favourite is I can be really lazy as I type code, and with format on save enabled I just press save and watch the code become well structured.
+
+
+## Grainformat
+
+The formatter can be invoked using the Grain cli
+
+```sh
+grain format myfile.gr
+```
+
+or 
+
+```sh
+grain format  
+``` 
+
+which will read from stdin.
+
+Both write the reformatted code to stdout.
+
+Coming soon will be a check mode and an in-place rewrite mode.
```

Comment:
```suggestion
which will read from stdin, and both write the reformatted code to stdout. In the near future, there will be a check mode and an in-place rewrite mode.
```
@marcusroberts maybe add a blurb about what the "check mode" will do?

Reply:
Thanks, done